### PR TITLE
Enable mask_starred in PSF iteration with JAX compatibility fixes

### PR DESCRIPTION
**Pull request by Guanlin-Liu**
This PR fixes several issues encountered when using the `starred` method

Changes:
- `lenstronomy/Data/psf.py`: fix numpy and JAX array type conflict
- `lenstronomy/Workflow/psf_fitting.py`: add PSF iteration mask support for starred 
- `lenstronomy/Sampling/Pool/multiprocessing.py`: fix deadlock caused by JAX and fork mode in multiprocessing (just a simple and straightforward approach)

### CODE REVIEW

**Comment by Guanlin-Liu** [comment]
Hi @martin-millon , I've opened this PR but found I can't choose reviewer from settings. So I’m afraid this is the only way I can inform you. Could you please review it? I'm very sorry to disturb you. Thanks!

**Comment by sibirrer** [comment]
@Guanlin-Liu thank you very much! I am not sure you can test the jax setting (see coverage report). If it's possible, that would be great.

File: lenstronomy/Data/psf.py [hunk 1]
```diff
@@ -117,7 +117,17 @@ def __init__(
                         "psf_variance_map are on the down-sampled pixel scale."
                     )
             if kernel_point_source_normalisation is True:
-                self._psf_variance_map /= np.sum(kernel_point_source) ** 2
+                # self._psf_variance_map /= np.sum(kernel_point_source) ** 2
+                # Handle the inconsistency between jax and numpy arrays for the psf variance map.
+                if isinstance(self._psf_variance_map, jax.Array) or isinstance(
```
Comment (ahuang314) [inline lenstronomy/Data/psf.py]: It might be better to just directly convert kernel_point_source and psf_variance_map to numpy arrays at the start, rather than performing this type-check. That way, an explicit JAX import is not necessary.

**Comment by Guanlin-Liu** [comment]
@sibirrer hi professor, I've added a JAX test using pytest.importorskip, so the test 
will be skipped if JAX is not available in the CI environment. 

The CI failure appears to be unrelated to my changes, it's caused by 
a compatibility issue between the docformatter/untokenize dependency 
and Python 3.14 in the CI environment.

**Comment by sibirrer** [comment]
Thank you very much @Guanlin-Liu! It looks like the CI fails on a line that you introduced: https://github.com/lenstronomy/lenstronomy/actions/runs/24614070880/job/71973915300?pr=837#step:5:510
The pre-commit part indeed, you can ignore (try to sync all the changes from main into your PR, perhaps that helps as well)

**Review by sibirrer — APPROVED** [review]
Thank you very much @Guanlin-Liu !
